src/main/java/net/minecraftforge/common/Dimension.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common;
+
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.registry.RegistryNamespaced;
+import net.minecraft.world.DimensionType;
+import net.minecraftforge.registries.GameData;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+
+public class Dimension extends IForgeRegistryEntry.Impl<Dimension>
+{
+    private final DimensionType type;
+    private final ResourceLocation dimID;
+    private int ticksWaited;
+    public static final RegistryNamespaced<ResourceLocation, Dimension> REGISTRY = GameData.getWrapper(Dimension.class);
+    
+    public Dimension(DimensionType type, String dimensionName)
+    {
+        this.type = type;
+        this.ticksWaited = 0;
+        dimID = this.setRegistryName(dimensionName).getRegistryName();
+        
+    }
+    
+    public static void registerDimensions()
+    {
+    	registerDimension(-1, new Dimension(DimensionType.NETHER, "minecraft:nether"));
+    	registerDimension(0, new Dimension(DimensionType.OVERWORLD, "minecraft:overworld"));
+    	registerDimension(1, new Dimension(DimensionType.THE_END, "minecraft:the_end"));
+
+    }
+
+    public static void registerDimension(Dimension dimension)
+    {
+    	REGISTRY.register(0, dimension.dimID, dimension);
```

Comment:
Remove these registration methods. The registry events are always getting fired after preinit. You can move the registry event somewhere else if you wish to,

Reply:
Don't we have to account for mods that create dimensions in the middle of the game though eg-RFTools, Mystcraft etc? This might not be the right way though.

Reply:
They can register dimensions in inits, but they can choose to load the level/world until the middle of the game.

Reply:
(Much like what multiworld plugins do)

Reply:
How do they know how many dimensions the player will create though?